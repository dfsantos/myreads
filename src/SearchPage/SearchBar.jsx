import React from 'react';
import PropTypes from 'prop-types';
import SearchInput from './SearchInput';

const propTypes = {
  onSearch: PropTypes.func.isRequired,
};

function SearchBar({ onSearch }) {
  return (
    <div className="search-books-bar">
      <a href="/" className="close-search">
        Close
      </a>
      <SearchInput onChange={onSearch} />
    </div>
  );
}

SearchBar.propTypes = propTypes;

export default SearchBar;
