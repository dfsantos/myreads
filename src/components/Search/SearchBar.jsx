import React from 'react';
import PropTypes from 'prop-types';

import ActionFlightTakeoff from 'material-ui/svg-icons/action/search';

import SearchInput from './SearchInput';

const propTypes = {
  onSearch: PropTypes.func.isRequired,
};

function SearchBar({ onSearch }) {
  return (
    <div className="search-books-bar">
      <div style={{ padding: 12 }}>
        <ActionFlightTakeoff />
      </div>
      <SearchInput onChange={onSearch} />
    </div>
  );
}

SearchBar.propTypes = propTypes;

export default SearchBar;
