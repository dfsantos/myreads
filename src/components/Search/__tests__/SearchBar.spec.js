import React from 'react';
import toJson from 'enzyme-to-json';

import SearchBar from '../SearchBar';

const onSearchEventListener = jest.fn();

it('should render without errors', () => {
  shallow(<SearchBar onSearch={onSearchEventListener} />);
});

it('should call onSearchEventListener when value change', () => {
  const wrapper = shallow(<SearchBar onSearch={onSearchEventListener} />);
  const input = wrapper.find('SearchInput');
  const event = { target: { value: 'android' } };

  input.simulate('change', event);

  expect(onSearchEventListener).toHaveBeenCalled();
});

it('default snapshot', () => {
  const wrapper = shallow(<SearchBar onSearch={onSearchEventListener} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
