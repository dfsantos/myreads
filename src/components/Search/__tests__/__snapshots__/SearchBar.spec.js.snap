// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default snapshot 1`] = `
<div
  className="search-books-bar"
>
  <div
    style={
      Object {
        "padding": 12,
      }
    }
  >
    <ActionSearch
      color="#ff6d00"
    />
  </div>
  <SearchInput
    onChange={[Function]}
  />
</div>
`;
